tests/unit_tests/test_source_mesh.py

```diff
@@ -351,51 +350,53 @@ def test_umesh_source_independent(run_in_tmpdir, request, void_model, library):
+    m1.add_nuclide('H1', 1.0)
+    m2 = m1.clone()
+    sph = openmc.Sphere(r=100, boundary_type='vacuum')
+    box1 = openmc.model.RectangularParallelepiped(-1, 0, -1, 1, -1, 1)
+    box2 = openmc.model.RectangularParallelepiped(0, 2, -1, 1, -1, 1)
+    cell1 = openmc.Cell(fill=m1, region=-box1)
+    cell2 = openmc.Cell(fill=m2, region=-box2)
+    outer = openmc.Cell(region=-sph & (+box1 | +box2))
     model = openmc.Model()
+    model.geometry = openmc.Geometry([cell1, cell2, outer])
 
-    rect_prism = openmc.model.RectangularParallelepiped(
-        -5.0, 5.0, -5.0, 5.0, -5.0, 5.0, boundary_type='vacuum')
-
-    mat = openmc.Material()
-    mat.add_nuclide('H1', 1.0)
-
-    model.geometry = openmc.Geometry([openmc.Cell(fill=mat, region=-rect_prism)])
-    model.settings.particles = 1000
-    model.settings.batches = 10
-    model.settings.run_mode = 'fixed source'
-
+    # Define a mesh covering the two cells: the first mesh element contains
+    # cell1 (-1 < x < 0) and the second element contains cells2 (0 < x < 2)
     mesh = openmc.RegularMesh()
-    mesh.lower_left = (-1, -2, -3)
-    mesh.upper_right = (2, 3, 4)
-    mesh.dimension = (1, 1, 1)
+    mesh.lower_left = (-3., -1., -1.)
+    mesh.upper_right = (3., 1., 1.)
+    mesh.dimension = (2, 1, 1)
 
-    model.settings.source = openmc.MeshSource(mesh, [file_source])
+    # Define a mesh source with a randomly chosen probability
+    p = random.random()
+    src1 = openmc.IndependentSource(strength=p, constraints={'domains': [cell1]})
+    src2 = openmc.IndependentSource(strength=1 - p, constraints={'domains': [cell2]})
+    model.settings.source = openmc.MeshSource(mesh, [src1, src2])
 
+    # Finish settings and export
+    model.settings.particles = 100
+    model.settings.batches = 1
     model.export_to_model_xml()
 
-    openmc.lib.init()
-    openmc.lib.simulation_init()
-    sites = openmc.lib.sample_external_source(10)
-    openmc.lib.simulation_finalize()
-    openmc.lib.finalize()
+    with openmc.lib.run_in_memory():
+        # Sample sites from the source
+        sites = openmc.lib.sample_external_source(N := 1000)
 
-    # The mesh bounds do not contain the point of the lone source site in the
-    # file source, so it should not appear in the set of source sites produced
-    # from the mesh source. Additionally, the source should be located within
-    # the mesh
-    bbox = mesh.bounding_box
-    for site in sites:
-        assert site.r != (0, 0, 0)
-        assert site.E == source_particle.E
-        assert site.u == source_particle.u
-        assert site.time == source_particle.time
-        assert site.r in bbox
+        # Check that all sites are either in cell1 or cell2
+        xs = np.array([s.r[0] for s in sites])
+        assert (xs >= -1.0).all()
+        assert (xs <= 2.0).all()
+
+        # Check that the correct percentage of the sites are in cell1
+        sigma = sqrt(p*(1- p)/N)
+        frac = xs[(-1.0 <= xs) & (xs <= 0.0)].size / N
```

Comment:
We could also check the source sites against the cell regions directly (`s.r in cell1.region ...`) but I think this is more efficient and clear enough.

Reply:
Not too worried about it, just noting it in case the test cells change in the future. Let's leave it as-is.